### PR TITLE
Create validator detail panel

### PR DESCRIPTION
**Pull request by ivanng-oursky**
/validators/likevaloper1sxjgwl2m35c6sah4f3yhd764csn307als4pm02?tab=voted&page=1

image

image

refs oursky/likedao#238

### CODE REVIEW

**Review by hochiw — CHANGES_REQUESTED** [review]
Screenshot 2022-07-15 at 2 55 31 PM

Seems to be a new button size, and different colour for the unstake button?

Screenshot 2022-07-15 at 2 56 54 PM

may need `justify-between` for APY container?

Screenshot 2022-07-15 at 2 58 26 PM

seems the spacing between the url and bonded tokens should be 24 (mb-6)?

Screenshot 2022-07-15 at 2 59 54 PM

logo becomes rectangle as I shrink the window

**Comment by ivanng-oursky** [comment]
@hochiw updated

https://user-images.githubusercontent.com/106721852/179202516-a3554950-8b65-449e-9210-b9df33ed74d4.mov

File: react-app/src/components/common/Buttons/AppButton.tsx [hunk 1]
```diff
@@ -48,7 +65,13 @@ function getButtonThemeClassNames(theme: AppButtonTheme): string {
         "disabled:bg-gray-300",
         "hover:bg-likecoin-lightgreen",
         "acitve:bg-likecoin-darkgreen",
-        "transition-colors"
+        "transition-colors",
+        "rounded-md",
+        "shadow-sm",
+        size === "extra-small" ? cn("py-2", "px-3") : null,
+        size === "small" ? cn("py-3", "px-4") : null,
+        size === "regular" ? cn("py-3", "px-6") : null,
```
Comment (hochiw) [inline react-app/src/components/common/Buttons/AppButton.tsx:73]: Seems very shared, should we keep the switch case function?

**Review by hochiw — CHANGES_REQUESTED** [review]

**Comment by ivanng-oursky** [comment]
@hochiw updated 🙇

File: react-app/src/components/common/Buttons/AppButton.tsx [hunk 2]
```diff
@@ -58,61 +92,64 @@ function getButtonThemeClassNames(theme: AppButtonTheme): string {
...
+        "border-likecoin-green",
+        "disabled:bg-gray-300",
+        "hover:text-white",
+        "transition-colors",
         "rounded-md",
-        "shadow-sm"
+        "shadow-sm",
+        size === "extra-small" ? cn("py-2", "px-3") : null,
+        size === "small" ? cn("py-3", "px-4") : null,
+        size === "regular" ? cn("py-3", "px-6") : null,
```
Comment (hochiw) [inline react-app/src/components/common/Buttons/AppButton.tsx:142]: getButtonPaddings?

File: react-app/src/components/common/Buttons/AppButton.tsx [hunk 3]
```diff
@@ -168,8 +205,8 @@ const AppButton = React.forwardRef<HTMLElement, AppButtonProps>(
     const computedClassName = useMemo(
       () =>
         cn(
-          getButtonThemeClassNames(theme),
-          getButtonSizeStyle(size),
+          getButtonClassNames(theme, size),
+          // getButtonSizeStyle(size),
```
Comment (hochiw) [inline react-app/src/components/common/Buttons/AppButton.tsx:209]: remove if not used

**Review by hochiw — CHANGES_REQUESTED** [review]

**Comment by ivanng-oursky** [comment]
@hochiw updated

**Review by hochiw — APPROVED** [review]

**Pull request merged**